mycore-base/src/main/java/org/mycore/common/content/transformer/MCRXSLTransformer.java

```diff
@@ -144,6 +149,18 @@ public static MCRXSLTransformer getInstance(String... stylesheets) {
         }
         return instance;
     }
+    
+    public static MCRXSLTransformer getInstance(Class<TransformerFactory> tfClass, String... stylesheets) {
+        String key = tfClass.getName() + "_"
+            + (stylesheets.length == 1 ? stylesheets[0] : Arrays.toString(stylesheets));
+        MCRXSLTransformer instance = INSTANCE_CACHE.get(key);
+        if (instance == null) {
+            instance = new MCRXSLTransformer(tfClass);
+            instance.setStylesheets(stylesheets);
+            INSTANCE_CACHE.put(key, instance);
+        }
```

Comment:
Use same cache handling for ALL getInstance() methods, e.g. call this method from `getInstance(String... stylesheets)`

Reply:
Sebastian and I agreed on having separate cache entries for initialisation with and without transformerfactory class